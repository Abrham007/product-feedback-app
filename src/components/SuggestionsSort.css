.SuggestionsSort {
  position: relative;
  margin-right: auto;
}

.SuggestionsSort__btn {
  display: flex;
  gap: 5px;
  align-items: center;
  background-color: transparent;
  border: none;
  outline: none;
  color: #f2f4fe;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: 400;
  line-height: normal;
  opacity: 0.75;
}

.SuggestionsSort__btn:hover {
  cursor: pointer;
}

.SuggestionsSort__drop-menu {
  background-color: white;
  position: absolute;
  top: 56px;
  left: 2px;
}

.SuggestionsSort__selected {
  font-weight: 700;
}

@media screen and (max-width: 768px) {
  .SuggestionsSort__btn {
    width: max-content;
    font-size: 0.8125rem;
    gap: 8px;
  }
}
