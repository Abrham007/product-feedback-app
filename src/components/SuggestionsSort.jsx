import { useState } from "react";
import DropMenu from "./DropMenu";
import "./SuggestionsSort.css";

function SuggestionsSort(props) {
  const [selectedIndex, setSelectedIndex] = useState(0);
  const [isOpen, setIsOpen] = useState(false);

  function toggleIsOpen() {
    setIsOpen((prevValue) => !prevValue);
  }

  function handleSelect(index) {
    setSelectedIndex(index);
    props.handleSort(index);
  }

  return (
    <div className="SuggestionsSort">
      <button onClick={toggleIsOpen} className="SuggestionsSort__btn">
        <span>Sort by :</span>
        <span className="SuggestionsSort__selected">
          {props.sortList[selectedIndex]}
        </span>
        {isOpen ? (
          <svg width="10" height="7" xmlns="http://www.w3.org/2000/svg">
            <path
              d="M1 6l4-4 4 4"
              stroke="#FFF"
              strokeWidth="2"
              fill="none"
              fillRule="evenodd"
            />
          </svg>
        ) : (
          <svg width="10" height="7" xmlns="http://www.w3.org/2000/svg">
            <path
              d="M1 1l4 4 4-4"
              stroke="#FFF"
              strokeWidth="2"
              fill="none"
              fillRule="evenodd"
            />
          </svg>
        )}
      </button>
      {isOpen && (
        <div className="SuggestionsSort__drop-menu">
          <DropMenu
            menuList={props.sortList}
            btnAction={handleSelect}
            selectedIndex={selectedIndex}
            toggleIsOpen={toggleIsOpen}
          />
        </div>
      )}
    </div>
  );
}

export default SuggestionsSort;
