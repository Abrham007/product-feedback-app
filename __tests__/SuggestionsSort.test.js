import SuggestionsSort from "../src/components/SuggestionsSort";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";

const mockHandleSort = jest.fn();
const sortList = [
  "Most Upvotes",
  "Least Upvotes",
  "Most Comments",
  "Least Comments",
];

describe("SuggestionsSort", () => {
  let component, tree;
  beforeEach(() => {
    component = renderer.create(
      <SuggestionsSort sortList={sortList} handleSort={mockHandleSort} />
    );
    tree = component.toJSON();
  });
  it("renders the correct JSX when first mounted ", () => {
    expect(tree).toMatchSnapshot();
  });

  it("renders the correct JSX after user clickes btn", () => {
    let root = component.root;
    renderer.act(() => {
      root.findByType("button").props.onClick();
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("sends the right choice to the parent", () => {
    render(<SuggestionsSort sortList={sortList} handleSort={mockHandleSort} />);
    fireEvent.click(screen.getAllByRole("button")[0]);
    fireEvent.click(screen.getAllByRole("button")[2]);
    expect(mockHandleSort.mock.calls).toHaveLength(1);
    expect(mockHandleSort.mock.calls[0][0]).toBe(1);
  });
});
