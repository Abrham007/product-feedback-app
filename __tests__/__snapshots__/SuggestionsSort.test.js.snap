// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionsSort renders the correct JSX after user clickes btn 1`] = `
<div
  className="SuggestionsSort"
>
  <button
    className="SuggestionsSort__btn"
    onClick={[Function]}
  >
    <span>
      Sort by :
    </span>
    <span
      className="SuggestionsSort__selected"
    >
      Most Upvotes
    </span>
    <svg
      height="7"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 6l4-4 4 4"
        fill="none"
        fillRule="evenodd"
        stroke="#FFF"
        strokeWidth="2"
      />
    </svg>
  </button>
  <div
    className="SuggestionsSort__drop-menu"
  >
    <menu
      aria-live="polite"
      className="DropMenu"
    >
      <li
        className="DropMenu__item"
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <button
          className="DropMenu__btn"
          onClick={[Function]}
        >
          <span>
            Most Upvotes
          </span>
          <img
            alt="check icon"
            aria-live="polite"
            className="DropMenu__icon"
            height={7.5}
            style={
              {
                "visibility": "visible",
              }
            }
            width={11}
          />
        </button>
      </li>
      <li
        className="DropMenu__item"
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <button
          className="DropMenu__btn"
          onClick={[Function]}
        >
          <span>
            Least Upvotes
          </span>
          <img
            alt="check icon"
            aria-live="polite"
            className="DropMenu__icon"
            height={7.5}
            style={{}}
            width={11}
          />
        </button>
      </li>
      <li
        className="DropMenu__item"
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <button
          className="DropMenu__btn"
          onClick={[Function]}
        >
          <span>
            Most Comments
          </span>
          <img
            alt="check icon"
            aria-live="polite"
            className="DropMenu__icon"
            height={7.5}
            style={{}}
            width={11}
          />
        </button>
      </li>
      <li
        className="DropMenu__item"
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <button
          className="DropMenu__btn"
          onClick={[Function]}
        >
          <span>
            Least Comments
          </span>
          <img
            alt="check icon"
            aria-live="polite"
            className="DropMenu__icon"
            height={7.5}
            style={{}}
            width={11}
          />
        </button>
      </li>
    </menu>
  </div>
</div>
`;

exports[`SuggestionsSort renders the correct JSX when first mounted  1`] = `
<div
  className="SuggestionsSort"
>
  <button
    className="SuggestionsSort__btn"
    onClick={[Function]}
  >
    <span>
      Sort by :
    </span>
    <span
      className="SuggestionsSort__selected"
    >
      Most Upvotes
    </span>
    <svg
      height="7"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1l4 4 4-4"
        fill="none"
        fillRule="evenodd"
        stroke="#FFF"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;
